What's New In DevTools (Microsoft Edge 92) (#1263)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/whats-new/2021/05/devtools.md

## The Close button is no longer hidden when DevTools is narrow

<!-- Title: DevTools is now easier to close -->  
<!-- Subtitle: The Close button in DevTools is always displayed, even when DevTools is docked to the right and the DevTools viewport is narrow. -->  

In Microsoft Edge version 91 or earlier, the **Close** button to close DevTools isn't displayed when the DevTools viewport is narrow.  In Microsoft Edge version 92, the **Close** button in the DevTools is always present, regardless of the DevTools viewport width.

:::image type="complex" source="../../media/2021/05/close-devtools-button-always-displayed.msft.png" alt-text="The Close DevTools button is now present even when the viewport is narrow" lightbox="../../media/2021/05/close-devtools-button-always-displayed.msft.png":::
   The **Close** DevTools button is now present even when the viewport is narrow  
:::image-end:::  

## Add tools quickly with the new More Tools button

<!-- Title: Add tools quickly with the new More Tools button -->  
<!-- Subtitle: Learn about a new convenient way to open tools in Microsoft Edge DevTools. -->  

> There's a new way to open more tools in Microsoft Edge DevTools: the **More Tools** (`+`) menu. The **More Tools** menu appears on the upper toolbar and on the lower toolbar (the toolbar of the drawer). Selecting a tool from the **More Tools** menu adds the tool to the toolbar.

Author: @Reezaali re: UI naming

To reorder the tabs on either toolbar, select and drag the tabs.

:::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button" lightbox="../../media/2021/05/more-tools-button.msft.png":::
:::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button on the upper toolbar" lightbox="../../media/2021/05/more-tools-button.msft.png":::
   The **More Tools** button on the upper toolbar
:::image-end:::  

:::image type="complex" source="../../media/2021/05/more-tools-menu.msft.png" alt-text="The More Tools menu" lightbox="../../media/2021/05/more-tools-menu.msft.png":::
   The **More Tools** menu
:::image-end:::  

> The **More Tools** button was available as an experiment in Microsoft Edge versions 89 through 91, and is now always present, starting with Microsoft Edge version 92.

Author: ```suggestion
The **More Tools** menu was available as an experiment since Microsoft Edge version 89 and is now always present.
```

Prior to Microsoft Edge version 92, in the **Console**, assistive technologies such as screen readers didn't announce autocomplete suggestions or the results of evaluated expressions. This has been fixed now.

:::row:::
    :::column:::
        :::image type="complex" source="../../media/2021/05/screen-reader-support-in-console-autocomplete.msft.png" alt-text="In the Console, screen readers now announce the currently selected autocomplete suggestion" lightbox="../../media/2021/05/screen-reader-support-in-console-autocomplete.msft.png":::
           In the **Console**, screen readers now announce the currently selected autocomplete suggestion
        :::image-end:::
    :::column-end:::
    :::column:::
        :::image type="complex" source="../../media/2021/05/screen-reader-support-in-console-evaluated-expression.msft.png" alt-text="In the Console, screen readers now announce the result of an evaluated expression" lightbox="../../media/2021/05/screen-reader-support-in-console-evaluated-expression.msft.png":::
           In the **Console**, screen readers now announce the result of an evaluated expression
        :::image-end:::
    :::column-end:::
:::row-end:::
        

> For a related feature in Microsoft Edge version 89, navigate to [Assistive technologies now announce position and count of CSS suggestions][DevtoolsWhatsnew202101DevtoolsAssistiveAnnounce].

Author: ```suggestion
```
Deleting it, to preserve focus on v92.

## Microsoft Edge Developer Tools for Visual Studio Code version 1.1.8

The [Microsoft Edge Developer Tools for Visual Studio Code][VisualstudioMarketplaceMsEdgedevtoolsVscodeEdgeDevtools] extension version 1.1.8 for Microsoft Visual Studio Code has the following changes since the previous release.  Microsoft Visual Studio Code updates extensions automatically.  To manually update to version 1.1.8, navigate to [Update an extension manually][VisualstudioCodeDocsEditorExtensionGalleryUpdateExtensionManually].  

You can file issues and contribute to the extension on the [vscode-edge-devtools GitHub repo][GithubMicrosoftVscodeEdgeDevtools].  

> ### In-context documentation and UI to make it easier to use the DevTools extension

Author: Removed the two UI-steps sections, linked to github.io instead, which has them.

<!-- Title: In-context documentation and UI make it easier to get started using the Developer Tools extension -->  
<!-- Subtitle: The Microsoft Edge Developer Tools for Visual Studio Code extension now presents helpful text, buttons, and links, and opens a documentation page with guidance on how to get started. -->  

Version 1.1.8 of the [Microsoft Edge Developer Tools for Visual Studio Code][VisualstudioMarketplaceMsEdgedevtoolsVscodeEdgeDevtools] extension now features a simpler way to start a new instance of Microsoft Edge, by presenting instructions, buttons, links, and a documentation page to guide you.

> *  When you select the **Microsoft Edge Tools** button in the **Activity Bar** on the left side of Visual Studio Code, the **Microsoft Edge Tools: Targets** panel now presents explanatory text, buttons, and links to guide you, instead of a blank panel.

Author: ```suggestion
*  When you select the **Microsoft Edge Tools** button in the **Activity Bar** of Visual Studio Code, the **Microsoft Edge Tools: Targets** panel now presents explanatory text, buttons, and links to guide you, instead of a blank panel.
```

    :::image-end:::  

1.  In the **Side Bar**, select the **Launch Instance** button.  Microsoft Edge opens, showing a documentation page, which provides guidance and links for more information.  Also, in Visual Studio Code, DevTools opens.

    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-doc-page.msft.png" alt-text="The new documentation page that appears in the browser, about how to use Microsoft Edge Developer Tools for Visual Studio Code" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-doc-page.msft.png":::
       The new documentation page that appears in the browser, about how to use Microsoft Edge Developer Tools for Visual Studio Code
    :::image-end:::  

1.  In the address bar, navigate to the project that you want to debug using Microsoft Edge Developer Tools for Visual Studio Code.

#### The Side Bar now provides a UI instead of a blank panel, to help launch your project for debugging

The Side Bar now provides UI instead of a blank panel, to help you launch your project when you want to debug it. 

> Perform the following steps to launch your project. These steps assume you are in Visual Studio Code, with no folder opened.

Author: The panel is different depending on whether a folder is opened.  This was a significant point requiring clarification.

> 1.  In the **Activity Bar** on the left, select the **Microsoft Edge Tools** button.  In the **Side Bar**, the **Microsoft Edge Tools: Targets** panel appears, showing a single button (**Launch Instance**).

Author: ```suggestion
1.  In the **Activity Bar**, select the **Microsoft Edge Tools** button.  In the **Side Bar**, the **Microsoft Edge Tools: Targets** panel appears, showing a single button (**Launch Instance**).
```

    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-initial-side-bar.msft.png" alt-text="The Microsoft Edge Tools: Targets panel in the Side Bar of Visual Studio Code, showing the Launch Instance button" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-initial-side-bar.msft.png":::
       The **Microsoft Edge Tools: Targets** panel in the **Side Bar** of Visual Studio Code, showing the **Launch Instance** button
    :::image-end:::  

> 1.  In the **Microsoft Edge Tools: Targets** panel, select the **open a folder** link, and open your project folder.  The **Microsoft Edge Tools: Targets** panel now shows two buttons: **Launch Instance** and **Generate launch.json**.

Reviewer: I think this can just be the start of the steps in this section. I don't think we need the previous step.

Author: Keeping previous step, because the UI differs based on whether a folder is opened.

    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png" alt-text="Editing the URL in the launch.json file" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png":::
       Editing the URL in the `launch.json` file
    :::image-end:::

1.  In the `launch.json` file, change the URL from an empty string to a URL for your project.  Or leave the string empty, to show the documentation page.

1.  Save the `launch.json` file.

1.  In the **Side Bar**, select the **Launch Project** button.  Microsoft Edge opens, showing the URL that you specified.  Also, in Visual Studio Code, DevTools opens.

## Download the Microsoft Edge preview channels  

If you are on Windows, Linux, or macOS, consider using the [Microsoft Edge preview channels][MicrosoftEdgePreviewChannels] as your default development browser.  The preview channels give you access to the latest DevTools features.  

## Getting in touch with Microsoft Edge DevTools team  

> [!INCLUDE [contact DevTools team note](../../includes/contact-whats-new-note.md)]

Reviewer: I think we need to update this note and remove the original page and author links. We can add them back when we actually have announcements from the Chromium project

Author: File "contact-whats-new-note.md" doesn't exist in this repo.

Author: @Reezaali - need your clarification of what actions to do re: the include.

[GithubMicrosoftVscodeEdgeDevtools]: https://github.com/microsoft/vscode-edge-devtools "microsoft/vscode-edge-devtools | GitHub"  
[GithubMicrosoftVscodeEdgeDevtoolsPullxxx]: https://github.com/microsoft/vscode-edge-devtools/pull/xxx "Pull xxx: Lorem al Ipsum | GitHub"  

[MicrosoftEdgePreviewChannels]: https://www.microsoftedgeinsider.com/download "Microsoft Edge Preview Channels"  

[VisualstudioCodeDocsEditorExtensionGalleryUpdateExtensionManually]: https://code.visualstudio.com/docs/editor/extension-gallery#_update-an-extension-manually "Update an extension manually - Extension Marketplace | Visual Studio Code"  

[VisualstudioMarketplaceMsEdgedevtoolsVscodeEdgeDevtools]: https://marketplace.visualstudio.com/items?itemName=ms-edgedevtools.vscode-edge-devtools "Microsoft Edge Developer Tools for Visual Studio Code | Visual Studio Marketplace"  

[YoutubeEdgeStateOfThePlatform]: https://www.youtube.com/watch?v=sU0WRZ0kkNo "Microsoft Edge: State of the Platform | YouTube"

[CRIssuesList]: https://bugs.chromium.org/p/chromium/issues/list "Chromium bugs"  
[CRxxxxxxx]: https://crbug.com/xxxxxxx "Issue xxxxxxx: Lorem al Ipsum | Chromium bugs"  

> > [!NOTE]
> > Portions of this page are modifications based on work created and [shared by Google][GoogleSitePolicies] and used according to terms described in the [Creative Commons Attribution 4.0 International License][CCA4IL].  
> > The original page is found [here](https://developer.chrome.com/blog/new-in-devtools-xx) and is authored by [Jecelyn Yeen][JecelynYeen] \(Developer advocate, Chrome DevTools\).  
> 

Author: ```suggestion
```

## The Close button is no longer hidden when DevTools is narrow

<!-- Title: DevTools is now easier to close -->  
<!-- Subtitle: The Close button in DevTools is always displayed, even when DevTools is docked to the right and the DevTools viewport is narrow. -->  

In Microsoft Edge version 91 or earlier, the **Close** button to close DevTools isn't displayed when the DevTools viewport is narrow.  In Microsoft Edge version 92, the **Close** button in the DevTools is always present, regardless of the DevTools viewport width.

:::image type="complex" source="../../media/2021/05/close-devtools-button-always-displayed.msft.png" alt-text="The Close DevTools button is now present even when the viewport is narrow" lightbox="../../media/2021/05/close-devtools-button-always-displayed.msft.png":::
   The **Close** DevTools button is now present even when the viewport is narrow  
:::image-end:::  

## Add tools quickly with the new More Tools button

<!-- Title: Add tools quickly with the new More Tools button -->  
<!-- Subtitle: Learn about a new convenient way to open tools in Microsoft Edge DevTools. -->  

> There's a new way to open more tools in Microsoft Edge DevTools: the **More Tools** (`+`) button and menu.  The **More Tools** (`+`) button appears on the upper toolbar and also on the lower toolbar (the toolbar of the drawer).  Selecting a tool from the **More Tools** menu adds the tool to the toolbar.

Author: ```suggestion
There's a new way to open more tools in Microsoft Edge DevTools: the **More Tools** (`+`) button and menu.  The **More Tools** (`+`) button appears on the upper toolbar and also on the lower toolbar (the toolbar of the drawer).  Selecting a tool from the **More Tools** menu adds the tool to the toolbar.
There's a new way to open more tools in Microsoft Edge DevTools: the **More Tools** (+) menu. The **More Tools** menu appears on the upper toolbar and on the lower toolbar (the toolbar of the drawer). Selecting a tool from the **More Tools** menu adds the tool to the toolbar.
```

To reorder the tool tabs on either toolbar, select and drag the tool tabs.

> :::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button" lightbox="../../media/2021/05/more-tools-button.msft.png":::

Author: ```suggestion
:::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button" lightbox="../../media/2021/05/more-tools-button.msft.png":::
:::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button on the upper toolbar" lightbox="../../media/2021/05/more-tools-button.msft.png":::
```

>    The **More Tools** button

Author: ```suggestion
   The **More Tools** button on the upper toolbar
```

        :::image type="complex" source="../../media/2021/05/hovering-on-tool-tab-in-dark-theme.msft.png" alt-text="When hovering over another tool's tab in dark theme, the tab is formatted to reduce the likelihood of accidentally closing it" lightbox="../../media/2021/05/hovering-on-tool-tab-in-dark-theme.msft.png":::
           When hovering over another tool's tab in dark theme, the tab is formatted to reduce the likelihood of accidentally closing it.
        :::image-end:::
    :::column-end:::
    :::column:::
        :::image type="complex" source="../../media/2021/05/hovering-on-tool-tab-in-light-theme.msft.png" alt-text="When hovering over another tool's tab in light theme, the tab is formatted to reduce the likelihood of accidentally closing it" lightbox="../../media/2021/05/hovering-on-tool-tab-in-light-theme.msft.png":::
           When hovering over another tool's tab in light theme, the tab is formatted to reduce the likelihood of accidentally closing it.
        :::image-end:::
    :::column-end:::
:::row-end:::

These improvements are especially relevant for users of localized DevTools, in which the tabs may be narrower and easier to accidentally close.

> :::image type="complex" source="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png" alt-text="In localized DevTools, some tabs are narrower, so the reformatting was especially needed" lightbox="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png":::

Author: ```suggestion
:::image type="complex" source="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png" alt-text=" Localized DevTools with narrow tabs" lightbox="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png":::
Localized DevTools with narrow tabs
	:::image-end:::
```

> :::image type="complex" source="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png" alt-text="In localized DevTools, some tabs are narrower, so the reformatting was especially needed" lightbox="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png":::

Author: ```suggestion
:::image type="complex" source="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png" alt-text="Localized DevTools with narrow tabs" lightbox="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png":::
```

>    In localized DevTools, some tabs are narrower, so the reformatting was especially needed

Author: ```suggestion
   Localized DevTools with narrow tabs
```

:::image-end:::

We also made it easier to re-add a tool that you closed by adding a [More Tools menu](#add-tools-quickly-with-the-new-more-tools-button) to the main toolbar and drawer toolbar.

## Better support for screen readers in the Console

<!-- Title: Better screen reader support in the Console -->
<!-- Subtitle: Assistive technologies can now announce autocomplete suggestions and evaluated expressions in the Console. -->

> In Microsoft Edge version 91 or earlier, in the **Console**, assistive technologies such as screen readers did not announce autocomplete suggestions or the results of evaluated expressions.  In Microsoft Edge version 92, in the **Console**, screen readers now announce the currently selected autocomplete suggestion, and also now announce the result of an evaluated expression.

Author: Prior to Microsoft Edge version 92, in the **Console**, assistive technologies such as screen readers didn't announce autocomplete suggestions or the results of evaluated expressions. This has been fixed now.

       The **Microsoft Edge Tools: Targets** panel in the **Side Bar** of Visual Studio Code with the Launch Instance and Generate launch.json buttons
    :::image-end:::  
    
1.  In the **Side Bar**, select the **Generate launch.json** button.  A `launch.json` file is generated in the root of the project, and is opened for editing.

    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png" alt-text="Editing the URL in the launch.json file" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png":::
       Editing the URL in the `launch.json` file
    :::image-end:::

1.  In the `launch.json` file, change the URL from an empty string to a URL for your project.  Or leave the string empty, to show the documentation page.

1.  Save the `launch.json` file.

1.  In the **Side Bar**, select the **Launch Project** button.  Microsoft Edge opens, showing the URL that you specified.  Also, in Visual Studio Code, DevTools opens.

> For more information, navigate to [Microsoft Edge Developer Tools for Visual Studio Code][VisualstudioMarketplaceMsEdgedevtoolsVscodeEdgeDevtools].

Author: ```suggestion
```

---
description: The More Tools button, in-context documentation to get started using the DevTools extension, increased support for screen readers in the Console, and more.
title: What's new in DevTools (Microsoft Edge 92)
author: MSEdgeTeam
ms.author: msedgedevrel
ms.date: 05/25/2021
ms.topic: article
ms.prod: microsoft-edge
keywords: microsoft edge, web development, f12 tools, devtools
---
# What's New In DevTools (Microsoft Edge 92)  

[!INCLUDE [contact DevTools team note](../../includes/edge-whats-new-note.md)]  

> [!TIP]

> > The **Microsoft Build 2021** conference is during May 25-27.  Here's a video from Build about the updates to DevTools: [Microsoft Edge | State of the Platform][YoutubeEdgeStateOfThePlatform] - "Microsoft Edge brings a compelling and consistent platform and tools for developers.  As our legacy browsers phase out of support, Edge will soon be the only supported browser from Microsoft on Windows 10.  Join us to learn about the latest across the Edge platform, tools, and web apps."

Reviewer 2: Now in the past, adding "happened from May..."

## The Close button is no longer hidden when DevTools is narrow

<!-- Title: DevTools is now easier to close -->  
<!-- Subtitle: The Close button in DevTools is always displayed, even when DevTools is docked to the right and the DevTools viewport is narrow. -->  

In Microsoft Edge version 91 or earlier, the **Close** button to close DevTools isn't displayed when the DevTools viewport is narrow.  In Microsoft Edge version 92, the **Close** button in the DevTools is always present, regardless of the DevTools viewport width.

:::image type="complex" source="../../media/2021/05/close-devtools-button-always-displayed.msft.png" alt-text="The Close DevTools button is now present even when the viewport is narrow" lightbox="../../media/2021/05/close-devtools-button-always-displayed.msft.png":::
   The **Close** DevTools button is now present even when the viewport is narrow  
:::image-end:::  

## Add tools quickly with the new More Tools button

<!-- Title: Add tools quickly with new More Tools button -->  
<!-- Subtitle: A convenient way to open new tools in Microsoft Edge DevTools. -->  

> There's a new way to open more tools in Microsoft Edge DevTools: the **More Tools** (`+`) button and menu.  The **More Tools** (`+`) button appears on the upper toolbar and also on the lower toolbar (the toolbar of the drawer).  Selecting a tool from the **More Tools** menu adds the tool to the toolbar.

Author: Per docs team discussion, docs team plans to write:
There's a new way to open more tools in Microsoft Edge DevTools: the **More Tools** (`+`) menu. The **More Tools**  menu appears on the upper toolbar and on the lower toolbar (the toolbar of the drawer). Selecting a tool from the **More Tools** menu adds the tool to the toolbar.

> There's a new way to open more tools in Microsoft Edge DevTools: the **More Tools** (`+`) button and menu.  The **More Tools** (`+`) button appears on the upper toolbar and also on the lower toolbar (the toolbar of the drawer).  Selecting a tool from the **More Tools** menu adds the tool to the toolbar.

Author: I added the planned text as a separate Suggestion, so I'm resolving this conversation without clicking "Add suggestion to batch".

To reorder the tool tabs on either toolbar, select and drag the tool tabs.

> :::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button" lightbox="../../media/2021/05/more-tools-button.msft.png":::

Author: It would be nice to show the + on the drawer toolbar

Author: Updated the capture in local working branch.

Author: This "More Tools" scenario, with two + buttons, is simpler than the wavy underlines.  I just finished re-doing this single capture to show both + signs, works well.

> :::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button" lightbox="../../media/2021/05/more-tools-button.msft.png":::

Author: If image is unchanged, alt text = "The More Tools button on the upper toolbar".
If image is changed to show the + buttons on both toolbars, alt text = "The More Tools button on the upper and lower toolbars".

> :::image type="complex" source="../../media/2021/05/more-tools-button.msft.png" alt-text="The More Tools button" lightbox="../../media/2021/05/more-tools-button.msft.png":::

Author: Resolving b/c I entered another Suggestion without " is highlighted".

>    The **More Tools** button

Author: If image is unchanged, alt text = "The More Tools button on the upper toolbar".
If image is changed to show the + buttons on both toolbars, alt text = "The More Tools button on the upper and lower toolbars".

:::image-end:::  

:::image type="complex" source="../../media/2021/05/more-tools-menu.msft.png" alt-text="The More Tools menu" lightbox="../../media/2021/05/more-tools-menu.msft.png":::
   The **More Tools** menu
:::image-end:::  

> The **More Tools** button was available as an experiment in Microsoft Edge versions 89 through 91, and is now always present, starting with Microsoft Edge version 92.

Author: @Reezaali - Please clarify "ATF"

> The **More Tools** button was available as an experiment in Microsoft Edge versions 89 through 91, and is now always present, starting with Microsoft Edge version 92.

Reviewer 3: yep

> The **More Tools** button was available as an experiment in Microsoft Edge versions 89 through 91, and is now always present, starting with Microsoft Edge version 92.

Author: ```suggestion
The **More Tools** menu was available as an experiment in Microsoft Edge version 89, and is now always present.
```
That version of the wording is from docs team discussion.

## Hovering, selecting, and closing tools is more visually consistent

<!-- Title: Improvements to light and dark themes -->

> <!-- Subtitle: Hovering, selecting, and closing tools is more visually consistent across both the default light and dark themes. -->

Author: I won't click "Add suggestion to batch" yet, in case hxInt revises the Suggestion per Reeza's comment.

> <!-- Subtitle: Hovering, selecting, and closing tools is more visually consistent across both the default light and dark themes. -->

Reviewer 4: LGTM--Thanks!

In the main toolbar at the top of DevTools, the tabs for each tool have been reformatted, to reduce the chance of accidentally closing a tool.  The highlighting that appears when hovering on another tool or hovering on the **More Tools** (`+`) button now looks consistent with the highlighting for the currently selected tool.

On each tool's tab in the main toolbar, we added:
*  Spacing around the tab.
*  Spacing around the close (`x`) button in the tab.
*  A background color when hovering over the tab.
*  A tooltip for the close (`x`) button of the tab.

> *  Higher contrast for the close (`x`) button of the tab.

Author: After I Commit the batched Suggestions, I'll wordsmith to include the drawer toolbar in this What's New item, and Push that commit into this PR.

For example, when you are in the **Animations** tool and you hover over the **Network** tool's tab, the **Network** tab is now formatted to reduce the likelihood of accidentally closing it:

> 

Author: I plan to keep the image and sentence.

:::row:::
    :::column:::
        :::image type="complex" source="../../media/2021/05/hovering-on-tool-tab-in-dark-theme.msft.png" alt-text="When hovering over another tool's tab in dark theme, the tab is formatted to reduce the likelihood of accidentally closing it" lightbox="../../media/2021/05/hovering-on-tool-tab-in-dark-theme.msft.png":::
           When hovering over another tool's tab in dark theme, the tab is formatted to reduce the likelihood of accidentally closing it.
        :::image-end:::
    :::column-end:::
    :::column:::
        :::image type="complex" source="../../media/2021/05/hovering-on-tool-tab-in-light-theme.msft.png" alt-text="When hovering over another tool's tab in light theme, the tab is formatted to reduce the likelihood of accidentally closing it" lightbox="../../media/2021/05/hovering-on-tool-tab-in-light-theme.msft.png":::
           When hovering over another tool's tab in light theme, the tab is formatted to reduce the likelihood of accidentally closing it.
        :::image-end:::
    :::column-end:::

> :::row-end:::

Author: Plan: update captures and text and alt-text, to show before vs after, instead of dark vs light.

This clearer formatting is especially relevant for users of localized DevTools, in which the tabs may be narrower, as shown below:

:::image type="complex" source="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png" alt-text="In localized DevTools, some tabs are narrower, so the reformatting was especially needed" lightbox="../../media/2021/05/hovering-reduced-chance-of-closing-tab.msft.png":::
   In localized DevTools, some tabs are narrower, so the reformatting was especially needed

> :::image-end:::

Author: Planned alt text: "Localized DevTools with narrow tabs"
Alt text should use pattern "Foo" instead of "Image of foo" or "Screen capture of foo".

> :::image-end:::

Author: Resolving and entering a new Suggestion without "Image showing ".

We also made it easier to re-add a tool that you closed, by adding a [More Tools menu](#add-tools-quickly-with-the-new-more-tools-button) on the upper and lower toolbars.

To customize the theme for your DevTools, navigate to [Enable Dark Theme in Microsoft Edge DevTools][DevtoolsCustomizeDarktheme].

## Better support for screen readers in the Console

<!-- Title: Better screen reader support in the Console -->
<!-- Subtitle: Assistive technologies can now announce autocomplete suggestions and evaluated expressions in the Console. -->

In Microsoft Edge version 91 or earlier, in the **Console**, assistive technologies such as screen readers did not announce autocomplete suggestions or the results of evaluated expressions.  In Microsoft Edge version 92, in the **Console**, screen readers now announce the currently selected autocomplete suggestion, and also now announce the result of an evaluated expression.

> 

Reviewer 2: That's a lot of repetiton, how about:

Prior to Microsoft Edge version 92  in the **Console**, assistive technologies such as screen readers did not announce autocomplete suggestions or the results of evaluated expressions.  This has been fixed now.

Author: Plan: Use the proposed text, but with a contraction:
Prior to Microsoft Edge version 92 in the Console, assistive technologies such as screen readers didn't announce autocomplete suggestions or the results of evaluated expressions. This has been fixed now.

           In the **Console**, screen readers now announce the result of an evaluated expression
        :::image-end:::
    :::column-end:::
:::row-end:::
        
For a related feature in Microsoft Edge version 89, navigate to [Assistive technologies now announce position and count of CSS suggestions][DevtoolsWhatsnew202101DevtoolsAssistiveAnnounce].

## Microsoft Edge Developer Tools for Visual Studio Code version 1.1.8

The [Microsoft Edge Developer Tools for Visual Studio Code][VisualstudioMarketplaceMsEdgedevtoolsVscodeEdgeDevtools] extension version 1.1.8 for Microsoft Visual Studio Code has the following changes since the previous release.  Microsoft Visual Studio Code updates extensions automatically.  To manually update to version 1.1.8, navigate to [Update an extension manually][VisualstudioCodeDocsEditorExtensionGalleryUpdateExtensionManually].  

You can file issues and contribute to the extension on the [vscode-edge-devtools GitHub repo][GithubMicrosoftVscodeEdgeDevtools].  

> ### In-context documentation and UI to make it easier to use the DevTools extension

Author: 'contextual documentation' sounds like a tech term, and might be a net loss in clarity.  Plan: no change.

<!-- Title: In-context documentation and UI make it easier to get started using the Developer Tools extension -->  
<!-- Subtitle: The Microsoft Edge Developer Tools for Visual Studio Code extension now presents helpful text, buttons, and links, and opens a documentation page with guidance on how to get started. -->  

> Version 1.1.8 of the [Microsoft Edge Developer Tools for Visual Studio Code][VisualstudioMarketplaceMsEdgedevtoolsVscodeEdgeDevtools] extension now features a simpler way to start a new instance of Microsoft Edge, by presenting instructions, buttons, links, and a documentation page to guide you.

Author: this link is redundant with above

Author: Plan: keep this link, since the 3rd redundant link (at end of section) has been removed.

       The **Microsoft Edge Tools: Targets** panel in the **Side Bar** of Visual Studio Code, showing the **Launch Instance** button
    :::image-end:::  

1.  In the **Side Bar**, select the **Launch Instance** button.  Microsoft Edge opens, showing a documentation page, which provides guidance and links for more information.  Also, in Visual Studio Code, DevTools opens.

    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-doc-page.msft.png" alt-text="The new documentation page that appears in the browser, about how to use Microsoft Edge Developer Tools for Visual Studio Code" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-doc-page.msft.png":::
       The new documentation page that appears in the browser, about how to use Microsoft Edge Developer Tools for Visual Studio Code
    :::image-end:::  

1.  In the URL bar, navigate to the project that you want to debug using Microsoft Edge Developer Tools for Visual Studio Code.

> #### The Side Bar now provides a UI instead of a blank panel, to help launch your project for debugging

Author: Rendered page section:
https://review.docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/whats-new/2021/05/devtools?branch=pr-en-us-1263#the-side-bar-now-provides-a-ui-instead-of-a-blank-panel-to-help-launch-your-project-for-debugging
Check that page to decide whether to convert the text from the h4 into a lead-in sentence.

These steps assume you are in Visual Studio Code, with no folder opened.

> 1.  In the **Activity Bar** on the left, select the **Microsoft Edge Tools** button.  In the **Side Bar**, the **Microsoft Edge Tools: Targets** panel appears, showing a single button (**Launch Instance**):

Reviewer 3: ```suggestion
1.  In the **Activity Bar** on the left, select the **Microsoft Edge Tools** button.  In the **Side Bar**, the **Microsoft Edge Tools: Targets** panel appears, showing a single button (**Launch Instance**).
```

    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-initial-side-bar.msft.png" alt-text="The Microsoft Edge Tools: Targets panel in the Side Bar of Visual Studio Code, showing the Launch Instance button" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-initial-side-bar.msft.png":::
       The **Microsoft Edge Tools: Targets** panel in the **Side Bar** of Visual Studio Code, showing the **Launch Instance** button
    :::image-end:::  

> 1.  In the **Microsoft Edge Tools: Targets** panel, select the **open a folder** link, and open a folder, such as your own project.  The **Microsoft Edge Tools: Targets** panel now shows two buttons: **Launch Instance** and **Generate launch.json**:

Reviewer 3: ```suggestion
1.  In the **Microsoft Edge Tools: Targets** panel, select the **open a folder** link, and open your project folder.  The **Microsoft Edge Tools: Targets** panel now shows two buttons: **Launch Instance** and **Generate launch.json**.
```

>     :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-side-bar.msft.png" alt-text="The Microsoft Edge Tools: Targets panel in the Side Bar of Visual Studio Code" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-side-bar.msft.png":::
>        The **Microsoft Edge Tools: Targets** panel in the **Side Bar** of Visual Studio Code
>     :::image-end:::  

Reviewer 3: ```suggestion
    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-side-bar.msft.png" alt-text="The Microsoft Edge Tools: Targets panel in the Side Bar of Visual Studio Code with the Launch Instance and Generate launch.json buttons" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-side-bar.msft.png":::
       The **Microsoft Edge Tools: Targets** panel in the **Side Bar** of Visual Studio Code with the Launch Instance and Generate launch.json buttons
    :::image-end:::  
```

Author: todo: bold the UI items in the 2nd alt text.

    
1.  In the **Side Bar**, select the **Generate launch.json** button.  A `launch.json` file is generated in the root of the project, and is opened for editing.

>     :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png" alt-text="Editing the URL in the launch.json file" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png":::
>        Editing the URL in the `launch.json` file
>     :::image-end:::

Reviewer 3: ```suggestion
    :::image type="complex" source="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png" alt-text="Edit the URL in the launch.json file" lightbox="../../media/2021/05/edge-tools-extension-in-context-docs-launch-json-file.msft.png":::
       Edit the URL in the `launch.json` file
    :::image-end:::
```

Reviewer 2: LGTM

Author: @codepo8 Does your "LGTM" comment apply to the original word ('Editing'), or the suggested change ('Edit')?  
This conversation was Resolved, thus hiding the "Add suggestion to batch" UI and implying I should do nothing and keep the word 'Editing'.

Author: @Reezaali - Pls see this Conversation UI - the result is ambiguous.  Since the conversation was Resolved and thus hidden and the buttons dimmed (unavailable), the UI appears to be telling me to ignore your Suggestion and keep the text as-is.  If I want to incorp your Suggestion, I'll need to first Unresolve the conversation, to make the buttons available.

> 1.  In the `launch.json` file, change the URL from an empty string to a URL for your project.  Or leave the string empty, to show the documentation page.

Reviewer 3: ```suggestion
1.  In the `launch.json` file, change the URL from an empty string to a URL for your project.  You can leave the URL string empty to show the default documentation page described in the previous section.
```

Author: @codepo8 Are you recommending that I keep the existing text ("Or leave the string empty, to show the documentation page.")?  I think so, because you marked the conversation as "Resolved", which made the **Commit suggestion** and **Add suggestion to batch** buttons dimmed.  To click those buttons, I would need to first click **Unresolve conversation**.

1.  Save the `launch.json` file.

1.  In the **Side Bar**, select the **Launch Project** button.  Microsoft Edge opens, showing the URL that you specified.  Also, in Visual Studio Code, DevTools opens.

> For more information, navigate to [Microsoft Edge Developer Tools for Visual Studio Code][VisualstudioMarketplaceMsEdgedevtoolsVscodeEdgeDevtools].

Author: this link is redundant with the two links above

Reviewer 3: Agreed.

Author: Plan: delete this link.

<!-- 
## Announcements from the Chromium project  

[!INCLUDE [contact DevTools team note](../../includes/chromium-whats-new-note.md)]  

### Ipsum et Chromium  

Lorem al lorem et Chromium  To review the history of this feature in the Chromium open-source project, navigate to Issue [xxxxxxx][CRxxxxxxx].  

:::image type="complex" source="../../media/202x/xx/lorem-et-chromium.msft.png" alt-text="Ipsum et Chromium" lightbox="../../media/202x/xx/lorem-et-chromium.msft.png":::
   Ipsum et Chromium  
:::image-end:::   

> -->

Reviewer 3: Can we remove this template code?

Author: plan: remove that template code (the commented-out block).

Author: ```suggestion
```
